pandas/_libs/tslibs/timedeltas.pyx

```diff
@@ -471,9 +471,12 @@ def _binary_op_method_timedeltalike(op, name):
     # define a binary operation that only works if the other argument is
     # timedelta like or an array of timedeltalike
     def f(self, other):
-        if hasattr(other, 'delta') and not PyDelta_Check(other):
-            # offsets.Tick
-            return op(self, other.delta)
+        if hasattr(other, '_typ'):
```

Comment:
rather than repeating this code can you make a function?

Reply:
thats exactly what this PR is preliminary to, but with more than just these 3 lines.

Reply:
oh ok